### PR TITLE
feat: [CI-6678]: add github connector encrypted fields

### PR DESCRIPTION
**Pull request by abhinav-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/45009)

### CODE REVIEW

File: 933-ci-commons/src/main/java/io/harness/git/GitTokenRetriever.java [hunk 1]
```diff
@@ -170,4 +177,27 @@ private String getGitApiURL(String url) {
       return "https://" + domain + "/api/v3/";
     }
   }
+
+  // todo: Use static method from here: io.harness.encryption.FieldWithPlainTextOrSecretValueHelper
+  public char[] getValueFromPlainTextOrSecretRef(String plainText, SecretRefData secretRefData) {
```
Comment (harshjain12) [inline 933-ci-commons/src/main/java/io/harness/git/GitTokenRetriever.java:182]: Can you fix todo
Why are we duplicating code ?

**Review by abhinav-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45009)* status: 9 file(s) to review. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`933-ci-commons/src/main/java/io/harness/git/GitTokenRetriever.java` line 182 at r2](https://reviewable.io/reviews/harness/harness-core/45009#-NQJb9YzboxOEh4Bzt1j-r2-182:-NQRrkT58ggFjLqzAAQ6:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/42705f445bfbaa14c085327a1240f699ae1d6adb/933-ci-commons/src/main/java/io/harness/git/GitTokenRetriever.java#L182)):*
<details><summary><i>Previously, harshjain12 (Harsh Jain) wrote…</i></summary><blockquote>

Can you fix todo
Why are we duplicating code ?
</blockquote></details>

Done.

**Review by harshjain12 — DISMISSED** [review]
:lgtm:

> Reviewed 7 of 8 files at r1, 1 of 1 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45009#-:-NQimq5Y68gUwZ-u86zN:bnfp4nl)* status: Some of the checks are too old.

**Comment by abhinav-harness** [comment]
trigger feature-build

**Comment by abhinav-harness** [comment]
trigger ut2

**Review by harshjain12 — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45009#-:-NQmvRuJ2sqNcMeiDADr:bnfp4nl)* status: :shipit: complete! ✓

**Review by archit-harness — APPROVED** [review]
> Reviewed 7 of 8 files at r1, 1 of 1 files at r3, 1 of 1 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45009)* status: :shipit: complete! ✓

**Review by adivishy1 — APPROVED** [review]
> Reviewed 7 of 8 files at r1, 1 of 1 files at r3, 1 of 1 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45009)* status: :shipit: complete! ✓

**Pull request merged**